_pytest/fixtures.py

```diff
@@ -1068,6 +1068,9 @@ def parsefactories(self, node_or_obj, nodeid=NOTSET, unittest=False):
         self._holderobjseen.add(holderobj)
         autousenames = []
         for name in dir(holderobj):
+            # Skip properties before the following getattr executes them.
+            if isinstance(getattr(type(holderobj), name, None), property):
```

Comment:
It is technically possible to raise any exception from `__getattr__` implemented on metaclass. Eg:
```pycon
>>> getattr(fields.Fields.a, 'a', None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/fields/__init__.py", line 301, in __getattr__
    raise TypeError("Field %r is already specified as required." % name)
TypeError: Field 'a' is already specified as required.
```